### ISSUE TITLE
Printf test is not checking scientific notation correctly

### ISSUE THREAD

**Issue by Jeremy-Kemp** [KhronosGroup/OpenCL-CTS#675]
The C spec says the following:

> The E conversion specifier produces a number with E instead of e introducing the exponent. The exponent always contains **at least two digits**, and only as many more digits as necessary to represent the exponent.

Conformance has a few cases where we test scientific notation, but testing against the wrong value.

For example,
```C
    printf("%+#21.15E\n", 789456123.0);
```
expects a result of `+7.894561230000000E+8`. However, the result should be `+7.894561230000000E+08`.

**Comment by jlewis-austin** [comment]
Just to clarify, from the code it looks like the issue is not that it doesn't _allow_ the correct 2-digit answer, but that it doesn't _require_ it, is that correct? I'm looking at the snippet that skips leading zeroes before comparing to the reference.

**Comment by Jeremy-Kemp** [comment]
Your comment has alerted me to the fact that this issue is one of my own creation that I spotted once I began fixing #673 .

The existing printf behaviour is to have reference results with all leading zeros stripped from the exponent. The test then also strips leading zeros from the actual result read back from the device.

With my changes, the reference result now contains leading zeros, but the test was still stripping the leading zeros from the device. It was at this point I noticed the "bug" and raised this issue.

Now I actually think that stripping leading zeros from the reference and the device result is desirable behaviour, as there is no comment on how many leading zeros there may be.

I'll update #680 to strip leading zeros from the reference and the actual result, and then close this issue.
